### PR TITLE
[Wait for upstream release] Resolve #231 - add bindAddress server config option to configure optional local binding address, pass through to node-datachannel peer connection init.

### LINKED ISSUE #231: Add support for setting the node-datachannel (& libdatachannel) bindAddress option for UDP connection
More details on the `node-datachannel` enhancement to support setting bindAddress: https://github.com/murat-dogan/node-datachannel/issues/143

# Background

Some tools like fly.io require binding UDP to a specific `bindAddress` (in fly.io's case, `fly-global-services` instead of `127.0.0.1` due to how they handle UDP packet routing).

## Support in next `node-datachannel` version

`libdatachannel` supports this, and with `node-datachannel`'s next update with https://github.com/murat-dogan/node-datachannel/pull/144 - it will, too!

## Adding to `geckos.io`!

Creating this issue to track updating `node-datachannel` and allow passing that config option through `geckos.io`!

### PR DESCRIPTION
**Pull request by bcjordan**
**NOTE: Waiting for v0.4.1 of node-datachannel to swap out temporary git commit hash package.json import with the new version.**

Resolves #231.

### CODE REVIEW

File: packages/server/package.json [hunk 1]
```diff
@@ -45,7 +45,7 @@
   "dependencies": {
     "@geckos.io/common": "^2.2.3",
     "@yandeu/events": "0.0.6",
-    "node-datachannel": "0.4.0"
+    "node-datachannel": "git://github.com/murat-dogan/node-datachannel.git#040bf82bba3b11aa45732e24878e2948f6655de4"
```
Comment (bcjordan) [inline packages/server/package.json:48]: Per https://github.com/murat-dogan/node-datachannel/issues/143#issuecomment-1337184210, wait for `node-datachannel` release `v0.4.1` to update this and then can merge.

**Comment by bcjordan** [comment]
@yandeu thanks, will do! I may also be able to provide a working example `fly.toml` in a separate repo down the road.

I also realized the git commit hash npm package is insufficient to test / needs `node-datachannel`'s full cross-platform CI build for a next version to include the change. Will hold off until then.

**Comment by yandeu** [comment]
Looks like v0.4.1 is very soon ready.

**Comment by FostUK** [comment]
On some aws ec2 tiers you might need this too as you need to bind to the private ip address of the ec2 instead of localhost.

**Pull request merged**